es6-renderer.js

```diff
@@ -11,7 +11,21 @@ const getPartial = (path, cb = 'resolveNeutral') => {
   };
   return new Promise(findFile);
 };
-    
+
+const defineAllVariables = function(template, locals) {
+  const localsDef = locals;
+  const pattern = /\$\{[\s]*?([\w\d]+)[\s]*?[\}\?]/gm;
+
+  let matchArray;
+  while((matchArray = pattern.exec(template)) !== null) {
+    if (!locals[matchArray[1]]) {
```

Comment:
This could have some false positive side effects when evaluating the following: '', false, 0.